pkg/search/postgres/common.go

```diff
@@ -797,9 +844,11 @@ func RunDeleteRequestForSchema(schema *walker.Schema, q *v1.Query, db *pgxpool.P
 		return err
 	}
 
-	_, err = db.Exec(context.Background(), query.AsSQL(), query.Data...)
-	if err != nil {
-		return errors.Wrapf(err, "could not delete from %q", schema.Table)
-	}
-	return nil
+	return pgutils.RetryQuery(func() error {
+		_, err = db.Exec(context.Background(), query.AsSQL(), query.Data...)
+		if err != nil {
+			return errors.Wrapf(err, "could not delete from %q", schema.Table)
```

Comment:
Well actually you do wrap the error here, so the `==` checks or direct casts would already fail here.

Reply:
Yup, fixed the error handling